joint_state_broadcaster/doc/userdoc.rst

```diff
@@ -38,3 +38,18 @@ An example parameter file
 
 .. generate_parameter_library_default::
   ../src/joint_state_broadcaster_parameters.yaml
+
+
+Order of the joints in the message
+----------------------------------
+
+The order of the joints in the message can determined by 3 different parameter settings:
+
+1. No defined ``joints`` parameter and ``use_urdf_to_filter`` set to ``false``:
+    The order of the joints in the message is the same as the order of the joints' state interfaces registered in the resource manager. This is typically the order in which the hardware components are loaded and configured.
+
+2. No defined ``joints`` parameter and ``use_urdf_to_filter`` set to ``true``:
+    The order of the joints in the message is the same as the order of the joints in the URDF file, which is inherited from the loaded URDF model and independent of the order in the `ros2_control` tag.
+
+3. Defined ``joints`` parameter:
+    The order of the joints in the message is the same as the order of the joints in the ``joints`` parameter.
```

Comment:
We should extend this to explain what happens when the list is less than the total available joints list and what if it has joints not available in the system currently